BrainPortal/lib/cbrain_task_generators/templates/portal.rb.erb

```diff
@@ -261,7 +266,55 @@ class CbrainTask::<%= name %> < <%= descriptor['cbrain:inherits-from-class'] ||
     # Check that disables-inputs is not violated
     # If the parameter is active, the ones it disables must not be
 %    generateCheckerLoop.("disable")
-% end   
+% end
+%#
+%####
+%# Check for violations pertaining to parameter groups
+%# In particular, check that mutual exclusivity and one-is-required are satisfied 
+%#
+% unless noGroups
+%# Check if any groups require checking
+% getGroupsWith = lambda { |prop| groups.select{ |g| g[prop] }.map{ |g| ":'"+g["id"]+"'" }.join(", ") }
+% mutexGrps     = getGroupsWith.("mutually-exclusive") 
+% oneReqGrps    = getGroupsWith.("one-is-required")
+% hasMutex      = (mutexGrps != "")
+% hasOneReq     = (oneReqGrps != "")
+%#
+% if (hasMutex || hasOneReq)
+   
+    # Groups with the mutually exclusive or one-is-required
+    <%= 'mutexGroups = [%s]'  % mutexGrps  if hasMutex %>
+    <%= 'oneReqGroups = [%s]' % oneReqGrps if hasOneReq %>
+    # Mapping from groupId to members list
+%   writeLiteralMap.("gidToMbrs", gIdToMbrs)
+    # Mapping from groupId to group name
+    grpName = {<%= groups.map{ |g| ":'" + g["id"] + "\' => \'" + g["name"] + "'" }.join(", ") %>}
+%  if hasMutex
+    # Lambda for checking mutual exclusivity
+    isMutex = lambda { |gid| gidToMbrs[gid].select{ |m| ! isInactive.(m.to_sym) }.count <= 1 }
+%  end
+%  if hasOneReq
+    # Lambda for checking one-is-required (at least one is active)
+    hasOneActive = lambda { |gid| gidToMbrs[gid].select{ |m| ! isInactive.(m.to_sym) }.count > 0 }
+%  end
+    # Check for violations of group properties
+%  if hasMutex  
+    mutexGroups.each do |group|
+      errMsg = "violates group mutual exclusivity requirement"
+      params_errors.add(grpName[group],errMsg) unless isMutex.(group)
+    end
+%  end
+%  if hasOneReq
+    oneReqGroups.each do |group|
+      errMsg = "violates group one-is-required specification"
+      params_errors.add(grpName[group],errMsg) unless hasOneActive.(group)
+    end
+%  end
```

Comment:
I'm still unsure why the if statements can't be grouped in a single "if hasMutex" and a single "if hasOneReq".